File: day_1/solution_part_one.py
with open("input.txt", "r") as f:
    print(
        (
            lambda entries: list(
                filter(
                    lambda k: k,
                    [
                        i * j if i + j == 2020 and not i == j else 0
                        for i in entries
                        for j in entries
                    ],
                )
            )
        )(list(map(int, f.readlines())))[0]
    )

File: day-1/rust/solution_part_one.rs
fn main() -> Result<(), Box<dyn std::error::Error>> {
    let input = std::fs::read_to_string("../input.txt")?
        .lines()
        .map(|x| x.parse::<u32>())
        .collect::<Result<Vec<u32>, std::num::ParseIntError>>()?;
    Ok(println!(
        "{:?}",
        input
            .iter()
            .filter(|x| input.iter().any(|y| *x != y && *x + *y == 2020))
            .fold(1, |x, y| x * *y)
    ))
}
